arangod/Scheduler/SupervisedScheduler.cpp

```diff
@@ -249,20 +251,36 @@ SupervisedScheduler::SupervisedScheduler(application_features::ApplicationServer
                                arangodb_scheduler_medium_prio_queue_length{}),
                            _server.getFeature<arangodb::MetricsFeature>().add(
                                arangodb_scheduler_low_prio_queue_length{})} {
-  _queues[0].reserve(maxQueueSize);
-  _queues[1].reserve(fifo1Size);
-  _queues[2].reserve(fifo2Size);
-  _queues[3].reserve(fifo3Size);
+  _queues[0].queue.reserve(maxQueueSize);
+  _queues[1].queue.reserve(fifo1Size);
+  _queues[2].queue.reserve(fifo2Size);
+  _queues[3].queue.reserve(fifo3Size);
   TRI_ASSERT(fifo3Size > 0);
 }
 
 SupervisedScheduler::~SupervisedScheduler() = default;
 
-bool SupervisedScheduler::queueItem(RequestLane lane, std::unique_ptr<WorkItemBase> work) {
+bool SupervisedScheduler::queueItem(RequestLane lane, std::unique_ptr<WorkItemBase> work, bool bounded) {
```

Comment:
can this method be a noexcept or still there might be throwable memory exceptions?

Reply:
Actually that is a good point. I don't vote for making this method noexcept, but what will happen if we increase the queue's counter and then the pushing to the queue throws an exception?
In that case we will have increased the counter but will not have pushed an item to the queue, and the counter will not fix itself anymore.
I think we need to wrap the `queue.push(...)` into a try...catch and decrease the queue's counter again if we catch an exception and the queue was bounded.
@mpoeter: what do you think?

Reply:
Good point, I agree with this